craft_application/models/base.py

```diff
@@ -89,6 +90,12 @@ def to_yaml_file(self, path: pathlib.Path) -> None:
         with path.open("wt") as file:
             util.dump_yaml(self.marshal(), stream=file)
 
+    def to_yaml_string(self) -> str:
+        """Return this model as a YAML string."""
+        stream = io.StringIO()
+        util.dump_yaml(self.marshal(), stream=stream)
+        return stream.getvalue()
```

Comment:
This should be the same as:

```suggestion
        return util.dump_yaml(self.marshal())
```

since upstream `pyyaml` returns a string if `stream=None`

Reply:
oh I didn't know that! Nice thanks